capactio/capact
# Migrate GCS to a new location, change long-runing cluster project

## Description

Changes proposed in this pull request:

Renaming:
- [x] `capactio-awesome-charts` -> `capactio-stable-charts`
    - migrated manually. I also cleaned up index.yaml from ghost releases.
- [x] `capactio-master-charts` -> `capactio-latest-charts`
    - No migration, charts will be pushed on main build
- [x] `capactio_binaries` -> `capactio-binaries`
    - `projectvoltron_ocftool` and `projectvoltron_dbpopulator` were also migrated to `capactio-binaries`
    - release notes were updated
- [x] `projectvoltron_terraform` -> `capactio-terraform-modules`. Such modules were migrated manually:
    - https://storage.googleapis.com/projectvoltron_terraform/terraform.aws.elasticsearch.provision/0.1.0/module.tgz
    - https://storage.googleapis.com/projectvoltron_terraform/terraform.aws.rds.postgresql.provision/0.1.0/module.tgz
    - https://storage.googleapis.com/projectvoltron_terraform/terraform.gcp.cloudsql.postgresql.install/0.2.0/module.tgz 
- [x]  `projectvoltron_cluster_recreate` -> `capact-stage-cluster-recreate`. It is now used by `backend-config` for terraform scripts in recreate cluster job.
- [x] `dev.cluster.projectvoltron.dev` -> `stage.cluster.capact.dev`. Currently configured in Cloud DNS in old project. We need to wait with that for Paul.

GitHub Secrets:
- [x] Change `GCS_CREDS` to `CAPACT_GCS_CREDS` 
- [x] Change `GKE_CREDS to `CAPACT_GKE_CREDS` 
- [x] Change `LONG_RUNNING_GATEWAY_PASSWORD` to `STAGE_CAPACT_GATEWAY_PASSWORD` 
- [x] **After merge,** I will delete `GCR_CREDS`, `GCS_CREDS`, `GKE_CREDS`, `LONG_RUNNING_GATEWAY_PASSWORD`

Load balancer IP:

Recreate cluster job:
- [x] Rework terraform apply in recreate cluster job to use GCS backend - in order to be able to destroy resources during failed apply
- [x] Change all suffix from `-dev` to `-stage` 
- [x] Add option to specify `overrideDockerRepository`. Useful when you need to run this job with images from PR.
- [x] Use secure gateway password instead of default one.
- [x] Create new external IP in `capact` project `34.78.197.21`
- [x] **After merge,** I will delete old GKE infrastructure and external IP.

Others:
-  Configure K8s API: 
   ```
   │ Error: googleapi: Error 403: Kubernetes Engine API has not been used in project 314865102738 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/container.googleapis.com/overview?project=314865102738 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry., accessNotConfigured
   ```

**Testing**

- The recreate job is green (both the `delete cluster` and `create cluster` steps were executed): https://github.com/capactio/capact/runs/2716410508?check_suite_focus=true
- New cluster is available at https://gateway.stage.cluster.capact.dev/graphql (as for pass on priv)
- Check that new GCS exists: https://console.cloud.google.com/storage/browser?authuser=1&project=capact&prefix=
- Check than new cluster is created in capact project: https://console.cloud.google.com/kubernetes/list?authuser=1&project=capact
